packages/client/src/actions/executeAction.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { OntologyDefinition } from "@osdk/api";
+import { applyActionV2 } from "@osdk/gateway/requests";
+import type { ApplyActionRequestOptions } from "@osdk/gateway/types";
+import type { ClientContext } from "@osdk/shared.net";
+import { createOpenApiRequest } from "@osdk/shared.net";
+import type { Actions } from "../types/Actions.js";
+
+export async function executeAction<
+  O extends OntologyDefinition<any>,
+  A extends keyof O["actions"],
+  Op extends ApplyActionRequestOptions,
+>(
+  client: ClientContext<OntologyDefinition<any>>,
+  actionApiName: A,
+  parameters?: Parameters<Actions<O>[A]>[0],
+  options?: Op,
+): Promise<void> {
+  const response = await applyActionV2(
+    createOpenApiRequest(client.stack, client.fetch),
+    client.ontology.metadata.ontologyApiName,
+    actionApiName as string,
+    {
+      parameters: parameters as any,
```

Comment:
Can we actually get away without having to marshal all the types to the wire format like in v1.1?

Reply:
Yeah we will need to handle these, but I don't think we've decided on the what attachments, timestamps, and dates will look like yet entirely. Objects and arrays of Objects wont work either I believe, and we need a way to extract the Object Set definition here as well. All the other primitives should be fine

Reply:
For most things I was just going to lean into raw types as long as possible. Good points on the edge cases